libdnf/dnf-context.cpp

```diff
@@ -3525,18 +3487,132 @@ dnf_context_module_install(DnfContext * context, const char ** module_specs, GEr
             messages.end(),std::make_move_iterator(solver_error.begin()), std::make_move_iterator(solver_error.end()));
     }
 
-    bool return_error = report_problems(messages);
-    if (return_error) {
-        g_set_error_literal(error, DNF_ERROR, DNF_ERROR_FAILED, _("Problems appeared for module install request"));
-        return FALSE;
-    } else {
-        for (const auto &nevra : nevras_to_install) {
-            if (!dnf_context_install (context, nevra.c_str(), error))
-              return FALSE;
+    // this code is based on dnf's ModuleBase.install()
+
+    g_autoptr(GPtrArray) resolved = g_ptr_array_new_with_free_func (g_free);
+    for (auto it = std::make_move_iterator(all_resolved_module_dicts.begin());
+              it != std::make_move_iterator(all_resolved_module_dicts.end()); it++) {
+        auto module_spec = std::get<0>(*it);
+        auto nsvcap_obj = std::get<1>(*it);
+        auto module_dict = std::get<2>(*it);
```

Comment:
I tried for a while to get `std::tie` to work here, but it simply doesn't like moving the Nsvcap object out of the tuple. We can look at this in a follow-up perhaps.